testpilot/frontend/static-src/app/main.js

```diff
@@ -59,22 +56,49 @@ app.extend({
         addEventListener('pageshow', app.router.history.checkUrl, false);
       }
     }).catch((err) => {
-      // for now, log the error in the console & do nothing in the UI
       console && console.error(err); // eslint-disable-line no-console
+      // HACK (for Issue #1101): Last ditch - reload the page and try again
+      window.location.reload();
     });
   },
 
   // Send webChannel message to addon, use a Promise to wait for the answer.
-  waitForMessage(type, data, timeout = 10000) {
+  waitForMessage(type, data, timeout = 5000, retries = 3) {
```

Comment:
FWIW, reduced timeout to 5 seconds from 10 - because the retries should actually lead to a total timeout of *15 seconds* before rejection.